Code/GraphMol/ChemTransforms/catch_tests.cpp

```diff
@@ -56,3 +58,202 @@ TEST_CASE("Github #1039", "[]") {
   }
 }
 
+TEST_CASE("molzip", "[]") {
+    SECTION("basic tests")
+    {
```

Comment:
I'd like to see some tests for what happens when things don't work (or don't get zipped together. For example:
```
        auto a = "C[*:1]"_smiles;
        auto b = "N[*:2]"_smiles;
        auto mol = molzip(*a,*b);
```
or, to be sure that things are actually being attached as you intend:
```
        auto a = "[*:2]OC[*:1]"_smiles;
        auto b = "N[*:1]"_smiles;
        auto mol = molzip(*a,*b);
```
or weird cases:
```
        auto a = "[*:1]OC[*:1]"_smiles;
        auto b = "N[*:1]"_smiles;
        auto mol = molzip(*a,*b);
```
and
```
        auto a = "[*:1]C[*:1]"_smiles;
        auto b = "N[*:1]"_smiles;
        auto mol = molzip(*a,*b);
```

Reply:
Right now I'm returning a null ptr, but that doesn't feel right.  I'll play with it.